mithril-core/src/merkle_tree.rs

```diff
@@ -123,7 +145,72 @@ impl<D: Digest + Clone> Path<D> {
     }
 }
 
-impl<D: Clone + Digest> MerkleTreeCommitment<D> {
+impl<D: Digest + FixedOutput> BatchPath<D> {
+    /// Convert the `BatchPath` into byte representation.
+    ///
+    /// # Layout
+    /// The layout of a `BatchPath` is
+    /// * Length of values
+    /// * Length of indices
+    /// * Values
+    /// * Indices
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut output = Vec::new();
+        let len_v = self.values.len();
+        let len_i = self.indices.len();
+
+        output.extend_from_slice(&u64::try_from(len_v).unwrap().to_be_bytes());
+        output.extend_from_slice(&u64::try_from(len_i).unwrap().to_be_bytes());
+
+        for value in &self.values {
+            output.extend_from_slice(value.as_slice())
+        }
+
+        for &index in &self.indices {
+            output.extend_from_slice(&u64::try_from(index).unwrap().to_be_bytes());
+        }
+        output
+    }
+
+    /// Try to convert a byte string into a `BatchPath`.
```

Comment:
Let's add this todo for future reference
```suggestion
    /// Try to convert a byte string into a `BatchPath`.
    // todo: We should not panic if the size of the slice is invalid (I believe `bytes[offset + i * 8..offset + (i + 1) * 8]` will panic if bytes is not large enough. 
```